lib/v8-to-istanbul.js

```diff
@@ -43,193 +41,257 @@ module.exports = class V8ToIstanbul {
       convertSourceMap.fromSource(rawSource) || convertSourceMap.fromMapFileSource(rawSource, dirname(this.path))
 
     if (rawSourceMap) {
-      if (rawSourceMap.sourcemap.sources.length > 1) {
-        console.warn('v8-to-istanbul: source-mappings from one to many files not yet supported')
-        this.source = new CovSource(rawSource, this.wrapperLength)
-      } else {
-        this._rewritePath(rawSourceMap)
-        this.sourceMap = await new SourceMapConsumer(rawSourceMap.sourcemap)
+      this.sourceMap = await new SourceMapConsumer(rawSourceMap.sourcemap)
+      this.sourceMapScanner = new SourceMapScanner(this.sourceMap)
+      const sourceRoot = rawSourceMap.sourcemap.sourceRoot
+
+      for (let i = 0; i < rawSourceMap.sourcemap.sources.length; i++) {
+        const originalPath = rawSourceMap.sourcemap.sources[i]
+        const resolvedPath = this._rewritePath(sourceRoot, originalPath)
 
         let originalRawSource
         // If the source map has inline source content then it should be here
         // so use this inline one instead of trying to read the file off disk
         // Not sure what it means to have an array of more than 1 here so just ignore it
         // since we wouldn't know how to handle it
-        if (this.sources.sourceMap && this.sources.sourceMap.sourcemap && this.sources.sourceMap.sourcemap.sourcesContent && this.sources.sourceMap.sourcemap.sourcesContent.length === 1) {
-          originalRawSource = this.sources.sourceMap.sourcemap.sourcesContent[0]
+        if (this.sources.sourceMap && this.sources.sourceMap.sourcemap && this.sources.sourceMap.sourcemap.sourcesContent[i]) {
+          originalRawSource = this.sources.sourceMap.sourcemap.sourcesContent[i]
         } else if (this.sources.originalSource) {
           originalRawSource = this.sources.originalSource
         } else {
-          originalRawSource = await readFile(this.path, 'utf8')
+          try {
+            originalRawSource = await readFile(resolvedPath, 'utf8')
+          } catch(e) {
```

Comment:
I needed to add this for me because my web app imported a package that imported a package and somewhere the source map path wasn’t made relative leading to failure.
You probably don’t want the console warn in - what are your preferences? An option to allow non found source files? Or an option to specify files to exclude?

Reply:
Specific file patterns to exclude would also improve performance by allowing excluding node modules.

Reply:
Ideally, I'd like to figure out a way to address the underlying bug, rather than suppressing an error ... I don't really want to introduce ignore logic in this layer, because there's already ignore logic the layer above in the [c8](https://github.com/bcoe/c8); If it's not easy to address the root cause, because it's a 3rd party dependency, maybe the tool calling `v8-to-istanbul` could do a bit of filtering?